### ISSUE TITLE
Simplified parallelism & cross-validation

### ISSUE THREAD

**Issue by iskandr** [hammerlab/mhcflurry#74]
Here's a sketch of a two-step cross-validation process where first we perform data preparation/imputation in parallel followed by parallel evaluation of the inner CV loop:

```
def transform_work(train_dataset, test_dataset):
    transformer = PrepareData()
    train2 = transformer.fit_transform(train_dataset)
    test2 = transformer.transform(test_dataset)
    return [Broadcast(train2), Broadcast(test2)]

 dataset_splits = kubeface.map(transform_work, cv_generator)

def cv_work(train_broadcast, test_broadcast, hyperparams):
        allele_to_model_dict = train_allele_specific_models(train_broadcast.value)
        allele_to_auc_dict = evaluate_aucs(models, test_broadcast.value)
        return (hyperparams, allele_to_model_dict, allele_to_auc_dict)

all_inputs = [
    (train_broadcast, test_broadcast, hyperparams) for 
    for (train_broacast, test_broadcast) in dataset_splits:
    for hyperparams in hyperparameter_generator]

all_results = kubeface.map(cv_work, all_inputs)

grouped_results = # group all_results by hyperparams objects into lists of AUC & model dicts
```

The assumption here is that all allele-specific work happens inside `train_allele_specific_models` and `evaluate_aucs` -- the CV code doesn't have to know about it. In order to achieve CV splits that are well balanced across both allele sizes and binding labels we should probably use a stratified iterator over a label that combines allele and IC50<=500nM.

**Comment by timodonnell** [comment]
No longer doing imputation so closing for now
